lib/beaker-hostgenerator/parser.rb

```diff
@@ -206,18 +207,18 @@ def parse_node_info_token(token)
     # @param host_settings [String] Non-nil user input string that defines host
     #                               specific settings.
     #
-    # @returns [Hash{String=>String}] The host_settings string as a map.
+    # @returns [Hash{String=>String|Array}] The host_settings string as a map.
     def settings_string_to_map(host_settings)
       # Strip it down to a list of pairs
       settings_pairs =
         host_settings.
         delete('{}').
-        split(',').
-        map { |keyvalue| keyvalue.split('=') }
-
+        split(/,(?=[^\]]*(?:\[|$))/)
```

Comment:
Could we update the comment to try to explain what this regex is doing in English?

Also looks like we could re-attach the `.map` to this chain of calls again.